sklearn/feature_selection/_mutual_info.py

```diff
@@ -62,12 +63,14 @@ def _compute_mi_cc(x, y, n_neighbors):
 
     # KDTree is explicitly fit to allow for the querying of number of
     # neighbors within a specified radius
-    kd = KDTree(x, metric="chebyshev")
-    nx = kd.query_radius(x, radius, count_only=True, return_distance=False)
+    kd = KDTree(x)
+    nx = kd.query_ball_point(x, radius, p=np.inf)
+    nx = [len(sub_list) for sub_list in nx]
     nx = np.array(nx) - 1.0
 
-    kd = KDTree(y, metric="chebyshev")
-    ny = kd.query_radius(y, radius, count_only=True, return_distance=False)
+    kd = KDTree(y)
+    ny = kd.query_ball_point(y, radius, p=np.inf)
+    ny = [len(sub_list) for sub_list in ny]
```

Comment:
Here and elsewhere in this diff, we can do even better (see below). It seems that the two APIs have effectively developed via convergent evolution to offer similar options with different names.

```diff
     kd = KDTree(y)
-    ny = kd.query_ball_point(y, radius, p=np.inf)
-    ny = [len(sub_list) for sub_list in ny]
+    ny = kd.query_ball_point(y, radius, p=np.inf, return_length=True)
```